pkg/mock/fileUtils_test.go

```diff
@@ -242,3 +242,63 @@ func TestFilesMockGlob(t *testing.T) {
 		assert.Len(t, matches, 0)
 	})
 }
+
+func TestStat(t *testing.T) {
+
+	var (
+		onlyMe                  os.FileMode = 0700
+		othersCanRead           os.FileMode = 0644
+		othersCanReadAndExecute os.FileMode = 0755
+	)
+
+	files := FilesMock{}
+	files.AddFile("tmp/dummy.txt", []byte("Hello SAP"))
+	files.AddDirWithMode("bin", 0700)
+
+	t.Run("non existing file", func(t *testing.T) {
+		_, err := files.Stat("doesNotExist.txt")
+		assert.EqualError(t, err, "stat doesNotExist.txt: no such file or directory")
+	})
+
+	t.Run("check file info", func(t *testing.T) {
+		info, err := files.Stat("tmp/dummy.txt")
+
+		if assert.NoError(t, err) {
+			t.Run("name", func(t *testing.T) {
+				// only the base name is returned.
+				assert.Equal(t, "dummy.txt", info.Name())
+			})
+			t.Run("dir", func(t *testing.T) {
+				assert.False(t, info.IsDir())
+			})
+			t.Run("mode", func(t *testing.T) {
+				// if not specified otherwise the 644 file mode is used.
+				assert.Equal(t, othersCanRead, info.Mode())
+			})
```

Comment:
Why wrap in more `t.Run()` calls and not simply additional asserts?

Reply:
That is  - of course - also possible. In personally prefer (in an ideal world) to have one assert per check. Reason: clearer test log output (`go test -v ./...`)

```
if assert.NoError(t, err) {
  // other check
}
```

does not count here.

In case there are other team pattern established over time this is also fine or me ...

Reply:
My usage pattern of running the tests may be different. I run all tests and want to see if any of them fail after having changed something. So once I see a failing test, I want to read the code and understand what the expectations are. So with these additional Run() calls, it just adds visual clutter in my usage. When I write tests myself, and the intention of an assert is not obvious, I extend it with a helpful message. My guess is that you can achieve the same thing by naming the Run() descriptively. But I think it just adds more code. I can't comment on the output of `go test -v ./...`, but my suspicion is that with a run name of "dir" or "mode", it is not very helpful by itself without looking at the code for the test anyway.

Reply:
Adjusted. I personally prefer more structured tests, even this ends up in more lines of code. But finally it is a matter of taste. Not important enough for fighting for it. --> Adjusted.